resource/graph_node.go

```diff
@@ -524,6 +524,14 @@ func (w *GraphNode) transitionTo(state NodeState) {
 		return
 	}
 
+	// if state of a node is removing it cannot transition to unhealthy until it is removed.
+	// currently this is the only hard blocked transition
+	// note this does not block SwapResource from transitioning a removing resource to ready
```

Comment:
minor, the substance of the comment is perfect, just clarifying that we are referencing a symbol. the brackets on `SwapResource` is a flyby that will create a link the generate docs.
 
```suggestion
	// if state of a node is [NodeStateRemoving] it cannot transition to [NodeStateUnhealthy] until it is removed.
	// currently this is the only hard blocked transition
	// note this does not block [SwapResource] from transitioning a removing resource to ready
```